### ISSUE TITLE
Missing top header icons

### ISSUE THREAD

**Issue by don570** [mypaint/mypaint#855]
#### Basic system details

MyPaint version:MyPaint-1.3.0-git.f7d69f5.glibc2.15-x86_64.AppImage 
Operating system: fatdog Linux  710   found at http://www.murga-linux.com/puppy/viewtopic.php?t=109067
Desktop environment: JWM + Rox filer + GTK2
I am using the  appimage version and it works well but I found this one bug
Missing  header icons at launch
I show a screenshot at
http://www.murga-linux.com/puppy/viewtopic.php?p=965204#965204

I launch with terminal  and here is output...
gdk_window_set_icon_list: icons too large is part of output
```
 # ./MyPaint-1.3.0-git.f7d69f5.glibc2.15-x86_64.AppImage 
zenity, kdialog, Xdialog missing. Skipping /tmp/.mount_MyPainEDDmaa/usr/bin//mypaint.wrapper.
INFO: mypaint: Installation layout: conventional POSIX-like structure with prefix u'/tmp/.mount_MyPainEDDmaa/usr'
INFO: lib.i18n: POSIX: LANG='en_US.UTF-8'
INFO: lib.i18n: POSIX: LANGUAGE=None

** (mypaint:24811): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files


New device 'Wacom Bamboo Pen eraser' (GDK_SOURCE_ERASER, axes:6, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Wacom Bamboo Pad pad' (GDK_SOURCE_PEN, axes:6, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Wacom Bamboo Pen stylus' (GDK_SOURCE_PEN, axes:6, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'HID 04b3:310d' (GDK_SOURCE_MOUSE, axes:3, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.device: New device 'Virtual core XTEST pointer' (GDK_SOURCE_MOUSE, axes:2, class=gtk.gdk.X11DeviceXI2, vendor='?', product='?')
INFO: gui.document: Initialized background from u'/root/.local/share/mypaint/backgrounds/default.png'

(mypaint:24811): Gdk-WARNING **: gdk_window_set_icon_list: icons too large
WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'
Traceback (most recent call last):
  File "/tmp/.mount_MyPainEDDmaa/usr/lib/mypaint/gui/tileddrawwidget.py", line 205, in _realize_cb
    win.set_event_compression(False)
AttributeError: 'gtk.gdk.X11Window' object has no attribute 'set_event_compression'
INFO: lib.document: load_ora: u'/root/.local/share/mypaint/scratchpads/autosave.ora'
INFO: lib.document: 0.519s load_ora total
INFO: gui.filehandling: Loaded scratchpad from u'/root/.local/share/mypaint/scratchpads/autosave.ora'

(mypaint:24811): Gtk-CRITICAL **: _gtk_tree_view_column_autosize: assertion 'GTK_IS_TREE_VIEW (tree_view)' failed

```

**Comment by achadwick** [comment]
Paging @probonopd (and assigning, as our appimage maintainer).

The gdk_window_set_icon_list() error message does not seem relevant.

When you try a normal install, do you get the icons?

**Comment by probonopd** [comment]
Since I am no Gtk+ expert by any stretch of the imagination, it would be good to hear from those who are.

**Comment by don570** [comment]
I  investigated  further and have tracked down  the icons that are causing the problem.
When I used  latest  appimage I noticed  that perspective  icon was showing  properly (see image)
![screenshot-icon](https://user-images.githubusercontent.com/19693002/30181522-8aa63a54-93e1-11e7-838c-63d7af9d5f97.png)

So I tracked  down where the icon images are stored.
/tmp/.mount_MyPainYJvjkc/usr/share/icons/hicolor/scalable/action/
I noticed that  most  SVG  images in folder are grey  whereas  the perspective icon image and a few others  are
black or colored. The black and colored ones will show  properly.

So while the  app was running  I made a copy of it to /opt/mypaint-1.3
to modify  the files.  I found that this  works!!!

I renamed  the offending SVG icons  so they won't be read (a different  extension).  Mypaint uses the 'file not found' icon now.
See image for an example....

![screenshot-renamed](https://user-images.githubusercontent.com/19693002/30181885-d794b4d4-93e2-11e7-87c5-2e5df25b791b.png)

The visibility icon for a layer is now useful  even though the eye icon isn't used.

I reported this also on fatdog64 site
http://www.murga-linux.com/puppy/viewtopic.php?p=966846#966846

**Comment by don570** [comment]
Does anyone  know  the  utility  app that reads SVG images that may be causing  the problem?

**Comment by odysseywestra** [comment]
That perspective tool is actually from a pull request. You probably need to go back a couple builds and test again.

**Comment by don570** [comment]
Thanks for  comment  but I think  you have  misunderstood.   Perspective tool works great
when I tested it last night.  

This  missing  icons issue  has been occurring  with other builds of mypaint  1.3
for  weeks now.  I wasn't the first  fatdog linux user to notice it.

The  screenshot image  I supplied above was  when  I deliberately changed the extension on the
perspective SVG image  I believe.  It was  just an experiment to see what would happen.
I should  have chosen  another icon  to change.

When I then changed the extension  on the visibility  SVG image I then saw an icon next to the  layer name  which is useful.  I will take  a screenshot  and show this behaviour.

**Comment by briend** [comment]
@don570 I saw you mentioned in a comment "Latest version of mypaint has perspective tool".  This is not the case.  There are several proposed changes- "pull requests" here:
https://github.com/mypaint/mypaint/pulls
Whenever one of these is updated it becomes "the latest" build on the CI server-- but it is not.

I think a lot of people are seeing the post here:
https://community.mypaint.org/t/where-to-download-windows-alpha-builds-of-mypaint/886
and there is some confusion.  Frankly, I'm confused too :-)

**Comment by don570** [comment]
I go to  https://travis-ci.org/mypaint/mypaint  for the image app  since I can't  do an installation otherwise.

**Comment by don570** [comment]
I was able to solve my problem   --->  Good enough  for my  purposes.

Here is  how I  did it.

I discovered that the  top icons  would  show  properly if  I made  24by24 icons and removed the 48by48
icons  (at least most of them).

As for   layer  visibility  and locking icons  -->
I found that I had to remove  the relevant  SVG  icons and install  22by22 icons.

For the  color  bookmark   --->  I removed  SVG icon  and installed 16by16 icon.

A lot of work  but I now can use the appimage in fatdog 710 linux
Here is link to download package.
http://www.murga-linux.com/puppy/viewtopic.php?p=967210#967210
